debug/gdbserver.py

```diff
@@ -697,8 +699,11 @@ def test(self):
             if value != tselect:
                 raise TestNotApplicable
             self.gdb.p(f"$tdata1=0x{tdata1:x}")
-            value = self.gdb.p("$tselect")
-            if value == tdata1:
+            value = self.gdb.p("$tdata1")
+            if value == tdata1 or \
+                    (value >> (xlen-4) == 0x6 and \
```

Comment:
`value >> (xlen-4) == 0x6` What is the purpose of this condition? If I read this right we explicitly ask for TYPE=2 trigger. How come we end up with TYPE=6?

Reply:
The purpose is to be compatible with spec 1.0 and 0.13. On our hardware that supports spec 0.13, setting the TYPE to 2  trigger actually we end up with TYPE=6.

Reply:
can we instead just re-request type==6 if the resulting value of tdata1 is different? I believe this would enable us to support wide range of platforms in a more cleaner way.